### ISSUE TITLE
Generated code incorrect for DBus structs in 0.0.12

### ISSUE THREAD

**Issue by Ace4896** [affederaffe/Tmds.DBus.SourceGenerator#6]
Hi, after upgrading from 0.0.11 to 0.0.12, I found that the generated code for structs seem to be incorrect. I've created a minimal repro [here](https://github.com/Ace4896/DbusGeneratedSignatureIssue) - `master` branch is on 0.0.11 and `generator-0.0.12` branch on 0.0.12.

It seems to be missing out the last field in structs, like these methods from `ReaderExtensions`:

```csharp
// Should be (ObjectPath, byte[], byte[], string)
public static (ObjectPath, byte[], byte[]) ReadStruct_roayaysz(this ref Reader reader)
{
    reader.AlignStruct();
    return ValueTuple.Create(reader.ReadObjectPath(), reader.ReadArray_ay(), reader.ReadArray_ay());
}

// Should be (ObjectPath, byte[], byte[])
public static (ObjectPath, byte[]) ReadStruct_roayayz(this ref Reader reader)
{
    reader.AlignStruct();
    return ValueTuple.Create(reader.ReadObjectPath(), reader.ReadArray_ay());
}
```

**Comment by affederaffe** [comment]
Seems like I messed with the SignatureReader which resulted in discarding the last element of structs. Should be fixed with 0.0.13, thanks!

**Comment by Ace4896** [comment]
Can confirm that it's fixed in 0.0.13, thanks for the quick update!
